proposals/p1964.md

```diff
@@ -97,33 +97,39 @@ like numeric literals:
 ## Details
 
 -   A character literal is a sequence enclosed with single quotes delimiter ('),
-    of UTF-8 code units that may or may not be a valid encoding. This matches
+    of UTF-8 code units that must be a valid encoding. This matches
     [the UTF-8 encoding of Carbon source files](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0142.md#character-encoding).
--   If a character literal encodes exactly one code point, then it supports
-    addition and subtraction. These operations produce another code point
-    literal, otherwise it will be rejected and produce an error.
--   Support for `\u` escape sequences.
--   Conversions from string literals to Unicode strings are implicit, even
-    though the numeric values of the encoding may change.
-
-    We will not support:
-
--   Character literals that don't contain exactly one Unicode code point.
--   New line and multi-line literals
--   "raw" literals (using #'x'#)
--   `\x` escape sequences
--   Single quote (`'`), but can write `'` character as `'\''`
--   Empty character literals (`''`)
--   ASCII Control codes (0...31), except when specified with an escape sequence.
--   Whitespace characters other than word space: tab, line feed, carriage
-    return, form feed, and vertical tab. And Back-slash (`\`), see
-    [String Literals: Escaping Sequence](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0199.md#escape-sequences-1)
+-   A character literal must encode exactly one code point.
+-   It supports addition and subtraction. These operations produce another code
+    point literal, or produce an error if the result is out of range.
+-   Character literals support some back-slash (`\`) escape sequences, including
+    `\t`, `\n`, `\r`, `\"`, `\'`, `\\`, `\0`, and `\u{HHHH...}`. See
+    [String Literals: Escape sequence](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0199.md#escape-sequences).
+-   Character literals implicitly convert to Unicode strings, even though the
+    numeric values of the encoding may change.
```

Comment:
@zygoloid Is there a situation where the numeric encoding will change when converting to a string now, or have our changes removed that possibility?

Reply:
I'm interpreting this as meaning: you can convert a character literal to a string, and the code point will be encoded in the encoding of the string (which I expect will be UTF-8). So, the encoding changes from UCS4 to UTF-8, but the encoded code point does not change. I think that's fine.